In [19]:
!git clone https://github.com/bhanu-50/Interview-assignment.git

fatal: destination path 'Interview-assignment' already exists and is not an empty directory.


In [20]:
import pandas as pd
import numpy as np

In [21]:
BonusCost = pd.read_excel('/content/Interview-assignment/Datasets/BonusCost_Data.xlsx')
First_Bet = pd.read_excel('/content/Interview-assignment/Datasets/First_Bet_Data.xlsx')
First_Deposit = pd.read_excel('/content/Interview-assignment/Datasets/First_Deposit_Data.xlsx')
Player_Activity = pd.read_excel('/content/Interview-assignment/Datasets/Player_Activity_Data.xlsx')
Player_Details = pd.read_excel('/content/Interview-assignment/Datasets/Player_Details.xlsx')

In [22]:
tables = [BonusCost,First_Bet,First_Deposit,Player_Activity,Player_Details]

In [23]:
BonusCost.head(5)

Unnamed: 0     Unnamed: 1           Unnamed: 2  Unnamed: 3  Unnamed: 4
0         NaN  Src_PLAYER_ID        ActivityMonth     Product  BONUS_COST
1         NaN       16051927  2020-06-30 00:00:00     eGaming          50
2         NaN       19309540  2021-02-28 00:00:00     eGaming          40
3         NaN       13176431  2020-01-31 00:00:00     eGaming          15
4         NaN       12405628  2019-02-28 00:00:00  SportsBook          30

In [24]:
First_Bet.head(5)

Unnamed: 0     Unnamed: 1                  Unnamed: 2  \
0         NaN  Src_Player_Id   System_First_Bet_Datetime   
1         NaN       15643431         2020-10-09 03:57:29   
2         NaN       13593034  2019-09-27 15:03:58.244000   
3         NaN       15894194         2021-02-28 02:22:07   
4         NaN       13238000         2019-07-08 10:39:45   

                 Unnamed: 3                      Unnamed: 4  \
0  System_First_BetSlip_Amt  System_First_Bet_Product_Group   
1                       2.5                         eGaming   
2                        10                          Sports   
3                         2                         eGaming   
4                       0.4                         eGaming   

                 Unnamed: 5                Unnamed: 6  \
0  System_First_Bet_Product  System_First_Bet_Channel   
1           Non Table Games                    Online   
2                  Prematch                    Online   
3           Non Table Games                    Online   
4           Non Table Games                    Online   

                  Unnamed: 7  
0  System_First_Bet_Platform  
1                     Mobile  
2                     Mobile  
3                        Web  
4                     Mobile

In [25]:
First_Deposit.head(5)

Unnamed: 0     Unnamed: 1          Unnamed: 2             Unnamed: 3  \
0         NaN  Src_Player_Id  First_Deposit_Date  First_Deposit_Channel   
1         NaN       13472812                 NaN                    NaN   
2         NaN       13331793                 NaN                    NaN   
3         NaN       13116242                 NaN                    NaN   
4         NaN       12689016                 NaN                    NaN   

             Unnamed: 4            Unnamed: 5  
0  First_Deposit_Method  First_Deposit_Amount  
1                   NaN                   NaN  
2                   NaN                   NaN  
3                   NaN                   NaN  
4                   NaN                   NaN

In [26]:
Player_Activity.head(5)

Unnamed: 0     Unnamed: 1           Unnamed: 2  Unnamed: 3  \
0         NaN  src_player_id        ActivityMonth     Product   
1         NaN       16585649  2020-08-31 00:00:00  SportsBook   
2         NaN       12845036  2019-05-31 00:00:00  SportsBook   
3         NaN       12720869  2021-06-30 00:00:00     eGaming   
4         NaN       18979743  2021-02-28 00:00:00     eGaming   

        Unnamed: 4        Unnamed: 5  Unnamed: 6  Unnamed: 7 Unnamed: 8  \
0       SubProduct  ActivePlayerDays  Bet_Amount  Win_Amount  Gross_Win   
1         Livebook                 1         500           0        500   
2         Prematch                28     1229.48     1177.85      41.63   
3  Non Table Games                 1        73.4        46.2       27.2   
4  Non Table Games                 1          52        21.7       30.3   

      Unnamed: 9  
0  Net_Gross_Win  
1            420  
2        34.9692  
3         22.848  
4         25.452

In [27]:
Player_Details.head(5)

Unnamed: 0     Unnamed: 1           Unnamed: 2           Unnamed: 3  \
0         NaN  Src_Player_Id          Signup_Date  acquisition_channel   
1         NaN       12584958  2019-02-23 17:24:49                  NaN   
2         NaN       12356925  2019-01-07 15:57:11                  NaN   
3         NaN       12500831  2019-02-10 08:31:13                  NaN   
4         NaN       12660009  2019-03-12 09:42:53                  NaN   

          Unnamed: 4 Unnamed: 5           Unnamed: 6          Unnamed: 7  
0  src_advertiser_id     Gender        Date_Of_Birth  Internal_Player_YN  
1             999999         M   1991-08-05 00:00:00                  N   
2             999999         M   1993-09-09 00:00:00                  N   
3             999999         M   1971-10-15 00:00:00                  N   
4             999999         M   2000-12-15 00:00:00                  N

In [28]:
First_Bet.head(5)

Unnamed: 0     Unnamed: 1                  Unnamed: 2  \
0         NaN  Src_Player_Id   System_First_Bet_Datetime   
1         NaN       15643431         2020-10-09 03:57:29   
2         NaN       13593034  2019-09-27 15:03:58.244000   
3         NaN       15894194         2021-02-28 02:22:07   
4         NaN       13238000         2019-07-08 10:39:45   

                 Unnamed: 3                      Unnamed: 4  \
0  System_First_BetSlip_Amt  System_First_Bet_Product_Group   
1                       2.5                         eGaming   
2                        10                          Sports   
3                         2                         eGaming   
4                       0.4                         eGaming   

                 Unnamed: 5                Unnamed: 6  \
0  System_First_Bet_Product  System_First_Bet_Channel   
1           Non Table Games                    Online   
2                  Prematch                    Online   
3           Non Table Games                    Online   
4           Non Table Games                    Online   

                  Unnamed: 7  
0  System_First_Bet_Platform  
1                     Mobile  
2                     Mobile  
3                        Web  
4                     Mobile

In [29]:
for df in tables:
  if df['Unnamed: 0'].isna().sum()==len(df):
    df.drop('Unnamed: 0', axis=1, inplace=True)
  df.columns = df.iloc[0]
  df = df[1:]
  df.reset_index(drop=True, inplace=True)

In [49]:
Player_Details.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 292786 entries, 0 to 292785
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   Src_Player_Id        292786 non-null  object
 1   Signup_Date          292786 non-null  object
 2   acquisition_channel  201280 non-null  object
 3   src_advertiser_id    290720 non-null  object
 4   Gender               292775 non-null  object
 5   Date_Of_Birth        292786 non-null  object
 6   Internal_Player_YN   292786 non-null  object
dtypes: object(7)
memory usage: 15.6+ MB


In [54]:
total_registrations = len(Player_Details[Player_Details['Signup_Date'].notna()])

In [68]:
total_registrations

292786

In [43]:
First_Deposit.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 292786 entries, 0 to 292785
Data columns (total 5 columns):
 #   Column                 Non-Null Count   Dtype 
---  ------                 --------------   ----- 
 0   Src_Player_Id          292786 non-null  object
 1   First_Deposit_Date     127262 non-null  object
 2   First_Deposit_Channel  127262 non-null  object
 3   First_Deposit_Method   127262 non-null  object
 4   First_Deposit_Amount   127262 non-null  object
dtypes: object(5)
memory usage: 11.2+ MB


In [48]:
total_first_deposit = len(First_Deposit[First_Deposit['First_Deposit_Date'].notna()])

In [67]:
total_first_deposit

127262

In [57]:
First_Bet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 292786 entries, 0 to 292785
Data columns (total 7 columns):
 #   Column                          Non-Null Count   Dtype 
---  ------                          --------------   ----- 
 0   Src_Player_Id                   292786 non-null  object
 1   System_First_Bet_Datetime       159167 non-null  object
 2   System_First_BetSlip_Amt        159167 non-null  object
 3   System_First_Bet_Product_Group  159167 non-null  object
 4   System_First_Bet_Product        159167 non-null  object
 5   System_First_Bet_Channel        159167 non-null  object
 6   System_First_Bet_Platform       159167 non-null  object
dtypes: object(7)
memory usage: 15.6+ MB


In [59]:
total_first_bet = len(First_Bet[First_Bet['System_First_Bet_Datetime'].notna()])

In [69]:
total_first_bet

159167

In [61]:
Player_Activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 860231 entries, 0 to 860230
Data columns (total 9 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   src_player_id     860231 non-null  object
 1   ActivityMonth     860231 non-null  object
 2   Product           860231 non-null  object
 3   SubProduct        860231 non-null  object
 4   ActivePlayerDays  860231 non-null  object
 5   Bet_Amount        860231 non-null  object
 6   Win_Amount        860231 non-null  object
 7   Gross_Win         860231 non-null  object
 8   Net_Gross_Win     860231 non-null  object
dtypes: object(9)
memory usage: 59.1+ MB


In [64]:
player_count_Active_in_first_30_days = len(Player_Activity[ pd.to_numeric(Player_Activity['ActivePlayerDays'], errors='coerce')>29])

In [65]:
total_registrations,total_first_deposit,total_first_bet,player_count_Active_in_first_30_days

(292786, 127262, 159167, 22624)

<h1><b>Conversion Funnel Analysis</b></h1>

<center>
<h3><b>Registrations → First Deposit</b></h3>


---

<br>
$$
\text{Conversion Percentage} = \frac{\text{First Deposits}}{\text{Registrations}} \times 100
$$
<br>
$$
\text{Conv %} = \frac{127262}{292786} \times 100
$$
<br>
$$
\text{Conv %} = 43.46\text{ }\%
$$

<br>

---


</center>



<center>
<h3><b>First Deposit → First Bet</b></h3>


---

<br>
$$
\text{Conversion Percentage} = \frac{\text{First Bet}}{\text{First Deposit}} \times 100
$$
<br>
$$
\text{Conv %} = \frac{159167}{127262} \times 100
$$
<br>
$$
\text{Conv %} = 125\text{ }\%
$$

<br>

---


</center>



<center>
<h3><b>First Bet → Active in first 30 days</b></h3>


---

<br>
$$
\text{Conversion Percentage} = \frac{\text{Active in first 30 days}}{\text{First Bet}} \times 100
$$
<br>
$$
\text{Conv %} = \frac{22624}{159167} \times 100
$$
<br>
$$
\text{Conv %} = 14.21\text{ }\%
$$

<br>

---


</center>



<h1><b>Insights</b></h1>

- Identify which stage has the largest drop-off and where (e.g., acquisition channel, cohorts)

In [72]:
df_upd = pd.merge(First_Deposit, First_Bet, on='Src_Player_Id', how='right')

In [73]:
df_upd.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 292786 entries, 0 to 292785
Data columns (total 11 columns):
 #   Column                          Non-Null Count   Dtype 
---  ------                          --------------   ----- 
 0   Src_Player_Id                   292786 non-null  object
 1   First_Deposit_Date              127262 non-null  object
 2   First_Deposit_Channel           127262 non-null  object
 3   First_Deposit_Method            127262 non-null  object
 4   First_Deposit_Amount            127262 non-null  object
 5   System_First_Bet_Datetime       159167 non-null  object
 6   System_First_BetSlip_Amt        159167 non-null  object
 7   System_First_Bet_Product_Group  159167 non-null  object
 8   System_First_Bet_Product        159167 non-null  object
 9   System_First_Bet_Channel        159167 non-null  object
 10  System_First_Bet_Platform       159167 non-null  object
dtypes: object(11)
memory usage: 24.6+ MB
